python/intro/README.md

```diff
@@ -0,0 +1,405 @@
+414666f7554d: Pull complete
+135a494fed80: Pull complete
+6ca3f38fdd4d: Pull complete
+4de6fcaa1241: Pull complete
+Digest: sha256:f76c4384f42bbb9065aff0a444c2d61f7ba3d9d44873976b4179b4e34280c6bb
+Status: Downloaded newer image for python:3.6
+ ---> 07d72c0beb99
+Step 2/6 : WORKDIR /function
+Removing intermediate container 9351c142b904
+ ---> 3ce33feb8e51
+Step 3/6 : ADD requirements.txt /function/
+ ---> f41ea90c9672
+Step 4/6 : RUN pip3 install -r requirements.txt
+ ---> Running in b20ba0ee4603
+Collecting fdk (from -r requirements.txt (line 1))
+  Downloading fdk-0.0.25-py2.py3-none-any.whl
+Collecting iso8601==0.1.12 (from fdk->-r requirements.txt (line 1))
+  Downloading iso8601-0.1.12-py2.py3-none-any.whl
+... # Truncated for brevity
+Step 5/6 : ADD . /function/
+ ---> 392b18c683d0
+Step 6/6 : ENTRYPOINT ["python3", "func.py"]
+ ---> Running in 29f238ca5b19
+Removing intermediate container 29f238ca5b19
+ ---> 2e4c259d3723
+Successfully built 2e4c259d3723
+Successfully tagged pythonfn:0.0.1
+{"message":"Hello World"}
+```
+
+You can also pass data to the run command. For example:
+
+![user input](images/userinput.png)
+>```sh
+> echo -n '{"name":"Bob"}' | fn run
+>```
+
+```yaml
+Building image fndemouser/pythonfn:0.0.1
+{"message":"Hello Bob"}
+```
+
+The JSON data was parsed and since `name` was set to "Bob", that value is passed
+in the output.
+
+### Understanding fn run
+If you have used Docker before the output of `fn --verbose run` should look
+familiar--it looks like the output you see when running `docker build`
+with a Dockerfile.  Of course this is exactly what's happening!  When
+you run a function like this Fn is dynamically generating a Dockerfile
+for your function, building a container, and then running it.
+
+> __NOTE__: Fn is actually using two images.  The first contains
+the language compiler and is used to generate a binary.  The second
+image packages only the generated binary and any necessary language
+runtime components. Using this strategy, the final function image size
+can be kept as small as possible.  Smaller Docker images are naturally
+faster to push and pull from a repository which improves overall
+performance.  For more details on this technique see [Multi-Stage Docker
+Builds for Creating Tiny Go Images](https://medium.com/travis-on-docker/multi-stage-docker-builds-for-creating-tiny-go-images-e0e1867efe5a).
+
+`fn run` is a local operation.  It builds and packages your function
+into a container image which resides on your local machine.  As Fn is
+built on Docker you can use the `docker` command to see the local
+container image you just generated.
+
+You may have a number of Docker images so use the following command
+to see only those created by fndemouser:
+
+![user input](images/userinput.png)
+>```sh
+> docker images | grep fndemouser
+>```
+
+You should see something like:
+
+```sh
+fndemouser/pythonfn   0.0.1               12f4f19c510c        About a minute ago   717MB
+```
```

Comment:
with a new image from CLI function will be 2 times smaller

Reply:
Fixed.